les/handler.go

```diff
@@ -361,6 +463,43 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 
 	var deliverMsg *Msg
 
+	errorFn := func(err error) {
+		if err != nil {
+			p.errCh <- err
+		}
+	}
+
+	sendReq := func(reqID, amount uint64, reply *reply, servingTime uint64) {
+		p.responseLock.Lock()
+		defer p.responseLock.Unlock()
+
+		cost := float64(servingTime)
+		inSizeCost := float64(msg.Size) * pm.server.inSizeCostFactor
+		if inSizeCost > cost {
+			cost = inSizeCost
+		}
+		if reply != nil {
+			outSizeCost := float64(reply.size()) * pm.server.outSizeCostFactor
+			if outSizeCost > cost {
+				cost = outSizeCost
+			}
+		}
```

Comment:
Please put the 'real cost' computation into it's own function or method and document that it is the maximum of 'serving time', 'input size' and 'output size'.